x/uibc/quota/keeper/quota.go

```diff
@@ -0,0 +1,276 @@
+package keeper
+
+import (
+	"encoding/json"
+	"strings"
+	"time"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	transfertypes "github.com/cosmos/ibc-go/v5/modules/apps/transfer/types"
+	"github.com/cosmos/ibc-go/v5/modules/core/exported"
+
+	ltypes "github.com/umee-network/umee/v4/x/leverage/types"
+	"github.com/umee-network/umee/v4/x/uibc"
+)
+
+// GetQuotaOfIBCDenoms returns quota of all tokens.
+func (k Keeper) GetQuotaOfIBCDenoms(ctx sdk.Context) ([]uibc.Quota, error) {
+	var quotaOfIBCDenoms []uibc.Quota
+
+	prefix := uibc.KeyPrefixDenomQuota
+	store := ctx.KVStore(k.storeKey)
+
+	iter := sdk.KVStorePrefixIterator(store, prefix)
+	defer iter.Close()
+
+	for ; iter.Valid(); iter.Next() {
+		var quotaOfIBCDenom uibc.Quota
+		if err := quotaOfIBCDenom.Unmarshal(iter.Value()); err != nil {
+			return nil, err
+		}
+
+		quotaOfIBCDenoms = append(quotaOfIBCDenoms, quotaOfIBCDenom)
+	}
+
+	return quotaOfIBCDenoms, nil
+}
+
+// GetQuotaByDenom retunes the rate limits of ibc denom.
+func (k Keeper) GetQuotaByDenom(ctx sdk.Context, ibcDenom string) (*uibc.Quota, error) {
+	return k.getQuotaByDenom(ctx, ibcDenom)
+}
+
+// getQuotaByDenom retunes the quota of ibc denom.
+func (k Keeper) getQuotaByDenom(ctx sdk.Context, ibcDenom string) (*uibc.Quota, error) {
```

Comment:
```suggestion
func (k Keeper) GetQuotaByDenom(ctx sdk.Context, ibcDenom string) (*uibc.Quota, error) {
```

Reply:
We can have one function, no need to have public which just calls the private and does nothing else